cmd/zetatool/filterdeposit/btc.go

```diff
@@ -0,0 +1,188 @@
+	return err
+}
+
+// getHashList is called by FilterBTCTransactions to help query and filter inbound transactions on btc
+func getHashList(cfg *config.Config, tssAddress string) ([]Deposit, error) {
+	var list []Deposit
+	lastHash := ""
+
+	// Setup URL for query
+	btcURL, err := url.JoinPath(cfg.BtcExplorerURL, "address", tssAddress, "txs")
+	if err != nil {
+		return list, err
+	}
+
+	// This loop will query the bitcoin explorer for transactions associated with the TSS address. Since the api only
+	// allows a response of 25 transactions per request, several requests will be required in order to retrieve a
+	// complete list.
+	for {
+		// The Next Query is determined by the last transaction hash provided by the previous response.
+		nextQuery := btcURL
+		if lastHash != "" {
+			nextQuery, err = url.JoinPath(btcURL, "chain", lastHash)
+			if err != nil {
+				return list, err
+			}
+		}
+		// #nosec G107 url must be variable
+		res, getErr := http.Get(nextQuery)
+		if getErr != nil {
+			return list, getErr
+		}
+
+		body, readErr := ioutil.ReadAll(res.Body)
+		if readErr != nil {
+			return list, readErr
+		}
+		closeErr := res.Body.Close()
+		if closeErr != nil {
+			return list, closeErr
+		}
+
+		// NOTE: decoding json from request dynamically is not ideal, however there isn't a detailed, defined data structure
+		// provided by blockstream. Will need to create one in the future using following definition:
+		// https://github.com/Blockstream/esplora/blob/master/API.md#transaction-format
+		var txns []map[string]interface{}
+		err := json.Unmarshal(body, &txns)
+		if err != nil {
+			return list, err
+		}
+
+		if len(txns) == 0 {
+			break
+		}
+
+		fmt.Println("Length of txns: ", len(txns))
+
+		// The "/address" blockstream api provides a maximum of 25 transactions associated with a given address. This
+		// loop will iterate over that list of transactions to determine whether each transaction can be considered
+		// a deposit to ZetaChain.
+		for _, txn := range txns {
+			// Get tx hash of the current transaction
+			hash := txn["txid"].(string)
+
+			// Read the first output of the transaction and parse the destination address.
+			// This address should be the TSS address.
+			vout := txn["vout"].([]interface{})
+			vout0 := vout[0].(map[string]interface{})
+			var vout1 map[string]interface{}
+			if len(vout) > 1 {
+				vout1 = vout[1].(map[string]interface{})
+			} else {
+				continue
+			}
+			_, found := vout0["scriptpubkey"]
+			scriptpubkey := ""
+			if found {
+				scriptpubkey = vout0["scriptpubkey"].(string)
+			}
+			_, found = vout0["scriptpubkey_address"]
+			targetAddr := ""
+			if found {
+				targetAddr = vout0["scriptpubkey_address"].(string)
+			}
+
+			//Check if txn is confirmed
+			status := txn["status"].(map[string]interface{})
+			confirmed := status["confirmed"].(bool)
+			if !confirmed {
+				continue
+			}
+
+			//Filter out deposits less than min base fee
+			if vout0["value"].(float64) < 1360 {
```

Comment:
This fee is calculated dynamically using `CalcDepositorFee` method.

Reply:
I'll need to get the actual block for each transaction in order to calculate that accurately. I will see how much it slows down.

Reply:
Unfortunately the api used here doesn't show the block height of the transaction, the block queries aren't showing lookups by tx id. For now this fee will have to be estimated until I can find a better API.

Reply:
we can adjust the base fee to something reasonable ... I will try to fix it later because this might require a completely different explorer.